src/System.Resources.Reader/src/System/Resources/ResourceReader.cs

```diff
@@ -329,10 +346,16 @@ private bool IsString(int typeIndex)
             return typeIndex == _stringTypeIndex;
         }
 
-        private static byte[] EncodeStringName()
+        private static byte[] EncodeStringName(bool fLegacyName)
         {
             var type = typeof(string);
             var name = type.AssemblyQualifiedName;
```

Comment:
We should just change this to a constant `var mscorlibName = "mscorlib, Version=";`